apps/test/unit/lib/ui/SchoolInfoInterstitialTest.jsx

```diff
@@ -50,16 +50,46 @@ describe('SchoolInfoInterstitial', () => {
             />
           </div>
           <div>
+            <Button
+              text={i18n.dismiss()}
+              onClick={wrapper
+                .find(Button)
+                .first()
+                .prop('onClick')}
+            />
             <Button
               text={i18n.save()}
-              onClick={wrapper.find(Button).prop('onClick')}
+              onClick={wrapper
+                .find(Button)
+                .last()
+                .prop('onClick')}
             />
           </div>
         </div>
       </BaseDialog>
     );
   });
 
+  it('closes the school info interstitial modal when the dismiss button is clicked', () => {
+    const wrapper = shallow(
+      <SchoolInfoInterstitial
+        {...MINIMUM_PROPS}
+        scriptData={{
+          ...MINIMUM_PROPS.scriptData,
+          existingSchoolInfo: {}
+        }}
+      />
+    );
+    const wrapperInstance = wrapper.instance();
+    sinon.spy(wrapperInstance, 'dismissSchoolInfoForm');
+    wrapper
+      .find(Button)
+      .first()
```

Comment:
Nit: I'm a little skeptical of using first()/last() in this file versus trying to find the dismiss button by id or something. Definitely a personal preference as I can see arguments both ways for this.

Reply:
Interestingly, I tried to accessing the button via an id but couldn't resolve this error:
``` 
Error: Method “simulate” is meant to be run on 1 node. 0 found instead.
```

Reply:
I don't think it's a big deal so I wouldn't try too hard.

`first()` seems better than `at(0)`. I was hopeful that there was a way to reference by id but I guess not.

Reply:
I tried .at(0) and .at(1) and got errors as well. :-)

Reply:
I know of a couple of solutions to this problem. [Enzyme's selector docs](https://airbnb.io/enzyme/docs/api/selector.html) are helpful in cases like this.

One is to use a special selector that combines the component's display name and a prop name/value filter.

```js
wrapper.find('Button[text="Dismiss"]')
// or 
wrapper.find('Button[id="dismiss-button"]')
```

Another, if you need maximum flexibility, is to use the wrapper [`filterWhere`](https://airbnb.io/enzyme/docs/api/ShallowWrapper/filterWhere.html) function with a predicate to select exactly the instance you need.

```js
wrapper
  .find(Button)
  .filterWhere(button => button.prop('id') === 'dismiss-button')
```

Reply:
@islemaster These are really 'neat' solutions.  I tried the solutions you recommended above and got the same error for each solution:

```AssertionError: expected dismissSchoolInfoForm to have been called at least once, but it was never called ```

It seems the button is not found.

Reply:
Oh weird, I actually tested these locally before posting them.  I wonder if one of us had dependencies out of sync - I'm sure the version of Enzyme has an impact on what works here and what doesn't.